### DISCUSSION TITLE
Facing issue with geometry values

### DISCUSSION THREAD

**Discussion by nagashreepoojari07** [OvertureMaps/data#333] [Discussion]
### Getting the below error while trying to download data in databricks.
**Error**: BinderException: Binder Error: No function matches the given name and argument types 'ST_GeomFromWKB(GEOMETRY)'. You might need to add explicit type casts.

**Code**:
import duckdb as duck
import pandas as pd
from shapely import wkb
db = duck.connect()
db.execute("INSTALL spatial")
db.execute("INSTALL httpfs")
db.execute("""
LOAD spatial;
LOAD httpfs;
SET s3_region='us-west-2';
""")

query = """
        select cast(ST_GeomFromWKB(geometry) as string) as wkt_geometry, id, geometry read_parquet('s3://overturemaps-us-west-2/release/2024-07-22.0/theme=divisions/type=division/*', filename=true, hive_partitioning=1) where bbox.xmin > -124.62738622773185 and bbox.xmax < -103.04854865820906 and bbox.ymin > 32.03683478447948 and bbox.ymax < 41.91410295376778;
    """

df = fetchDarafromOverturemaps(db, query)
display(df)
data = db.execute(query)
new_df = data.fetch_df()
display(new_df)

**geometry** value seems to be wrong, when same query executed 8 days back had different geometry values, attached screenshots of the data comparison.
**8 days back**
![before](https://github.com/user-attachments/assets/a6e37957-3ece-45d3-98f6-b2613a91da53)
**present**
![after](https://github.com/user-attachments/assets/401af5a9-4cb0-42f0-a240-37f55ab66d4b)

**Comment by nagashreepoojari07** [comment]
DuckDB-1.1.0 seems to be the issue,
works fine with version -1.0.0

**Comment by danabauer** [comment]
Thanks for highlighting this, @nagashreepoojari07. The [latest release of DuckDB (1.1.0)](https://duckdb.org/2024/09/09/announcing-duckdb-110) automatically converts geometry columns to the GEOMETRY type when reading GeoParquet files, so you don't need to cast the geometry in your query. We'll be updating all the DuckDB examples in our docs soon to reflect this new functionality.
